tests/framework/extensions/clusters/aks/resources/aks_cluster.go

```diff
@@ -0,0 +1,126 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+)
+
+var (
+	balanceSimilarNodeGroups = "false"
+	ctx                      = context.Background()
+	defaultRandStringLength  = 5
+	disableLocalAccounts     = false
+	dnsPrefix                = "auto-dns"
+	enableAutoScaling        = true
+	enablePublicIP           = false
+	enableRBAC               = true
+	k8sVersion               = "1.24.6"
```

Comment:
Is this a best practice to hard code all these values as defaults?  Perhaps some could be pulled from the config.yml to keep things more dynamic?

Reply:
Possibly. I made the `resources` folder with two particular goals in mind. The first was to eliminate overcrowding the config file with excessive parameters like it does now. The second is that these values are making an AKS cluster from Azure directly, so aside from the K8s version, the defined values should not be left up to the user to tinker with, otherwise the defined tests won't work.

Reply:
Yeah I should have Ugur's changes, I'll take a look. So, do we want to go with putting all values into the config file then? My main concern is excluding the K8s version, the values aren't supposed to change in order to populate an AKS cluster from Azure.

Reply:
Can you ask Ugur how he did it? I don't want hardcoded values especially here in the extensions, which are to be used by all tests, and these values are specific to a test case. So they definitely don't belong here. Can you see how we do upgrade tests for charts? We are essentially doing the same. What he do there is we make sure we create a chart with a version previous to the latest, that we ensure we can upgrade. That is much more efficient than hard coding it. The rest can be a config entry in a yaml in the actual test directory. This is to avoid having to do a whole PR just to update certain values, which I want to avoid.

Reply:
I agree with that, that'd be great if we could reduce hardcoded values. Charts and Kubernetes upgrades can be taken as examples, I am open to providing more information on them